From 1123ffc5969fbf8861f11e7e29d13361b5248d86 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Mon, 23 Jun 2025 19:19:37 +0800
Subject: [PATCH 14/63] loonggpu: Remove test for !screen_base in fbdev probing

The screen_base field comes from the fbdev BO and contains the fbdev
framebuffer base address. We get the BO memory via gsgpu_bo_kmap(),
which already reports the error -ENOMEM if the buffer could not be
mapped. So remove the later test for screen_base, which will never
be NULL at this point.

Following the reference implementation change.

Link: https://git.kernel.org/torvalds/c/11b6005865e4
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_fb.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/loonggpu/loonggpu_fb.c b/loonggpu/loonggpu_fb.c
index c0cca28..524e314 100644
--- a/loonggpu/loonggpu_fb.c
+++ b/loonggpu/loonggpu_fb.c
@@ -253,11 +253,6 @@ static int loonggpufb_create(struct drm_fb_helper *helper,
 
 	/* Use default scratch pixmap (info->pixmap.flags = FB_PIXMAP_SYSTEM) */
 
-	if (info->screen_base == NULL) {
-		ret = -ENOSPC;
-		goto out;
-	}
-
 	DRM_INFO("fb mappable at 0x%lX\n",  info->fix.smem_start);
 	DRM_INFO("vram apper at 0x%lX\n",  (unsigned long)adev->gmc.aper_base);
 	DRM_INFO("size %lu\n", (unsigned long)loonggpu_bo_size(abo));
-- 
2.52.0

